models/migrations/migrations.go

```diff
@@ -638,14 +640,16 @@ func attachmentRefactor(x *xorm.Engine) error {
 		log.Info("Failed to rename some attachments, old and new paths are saved into: %s", dumpPath)
 	}()
 	for _, attach := range attachments {
-		if err = os.MkdirAll(path.Dir(attach.NewPath), os.ModePerm); err != nil {
-			isSucceed = false
-			return err
-		}
+		if filepath.IsAbs(setting.AttachmentPath) {
```

Comment:
Why checking only for absolute attachment path?

Reply:
```
AttachmentPath	 = "/srv/attachments/"
```
If the AttachmentPath is absolute, prefix "file://" must be added to it in order to open it using `blob.OpenBucket`.
That's why I checked it.

Reply:
I have updated it. I've attached a summary how whole `BucketURL` and `AttachmentPath` or others work.
https://github.com/go-gitea/gitea/blob/4c390c9699ee6400a9467c4c8234af9efba5acaf/modules/setting/setting.go#L1053-L1083

Please take a look if it seems okay. Thanks